File: app/src/main/java/org/jeonfeel/moeuibit2/getUpBitAPI/Retrofit_UpBit.java
package org.jeonfeel.moeuibit2.getUpBitAPI;

import org.jeonfeel.moeuibit2.Activitys.Activity_coinDetails.SelectedCoinModel;

import java.util.List;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class Retrofit_UpBit {

    private Retrofit retrofit;

    public Retrofit_UpBit() {

        this.retrofit = new retrofit2.Retrofit.Builder()
                .baseUrl("https://api.upbit.com/")
                .addConverterFactory(GsonConverterFactory.create())
                .build();
    }

    public retrofit2.Retrofit getRetrofit_upBit() {

        return retrofit;
    }

    public retrofit2.Call<List<SelectedCoinModel>> getSelectedCoinCall(String market) {

        UpbitService upbitService = retrofit.create(UpbitService.class);

        retrofit2.Call<List<SelectedCoinModel>> call = upbitService.getCoin(market);

        return call;
    }
}

File: app/src/main/java/org/jeonfeel/moeuibit2/data/remote/retrofit/api/RetrofitUpBit.kt
package org.jeonfeel.moeuibit2.data.remote.retrofit.api

import com.google.gson.JsonArray
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.Call

class RetrofitUpBit {
    private val TAG = RetrofitUpBit::class.java.simpleName
    private val retrofit: Retrofit = Retrofit.Builder()
        .baseUrl("https://api.upbit.com/")
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    //Call 객체를 반환.
    fun getMarketCodeCall(): Call<JsonArray> {
        val upBitService =
            retrofit.create(UpBitService::class.java)
        return upBitService.getMarketCode()
    }

    fun getKrwTickerCall(markets: String): Call<JsonArray> {
        val upBitService =
            retrofit.create(UpBitService::class.java)
        return upBitService.getKrwTicker(markets)
    }
}